source/image-handler/image-handler.js

```diff
@@ -1,160 +1,197 @@
+    const edits = request.edits;
+    if (edits !== undefined) {
+      const modifiedImage = await this.applyEdits(originalImage, edits);
+      if (request.outputFormat !== undefined) {
+        await modifiedImage.toFormat(request.outputFormat);
+      }
+      const bufferImage = await modifiedImage.toBuffer();
+      return bufferImage.toString("base64");
+    } else {
+      return originalImage.toString("base64");
+    }
+  }
+
+  /**
+   * Applies image modifications to the original image based on edits
+   * specified in the ImageRequest.
+   * @param {Buffer} originalImage - The original image.
+   * @param {Object} edits - The edits to be made to the original image.
+   */
+
+  async applyEdits(originalImage, edits) {
+    const image = sharp(originalImage);
+    const keys = Object.keys(edits);
+    const values = Object.values(edits);
+    // Apply the image edits
+    for (let i = 0; i < keys.length; i++) {
+      const key = keys[i];
+      const value = values[i];
+      if (key === "overlayWith") {
+        const overlay = await this.getOverlayImage(value.bucket, value.key);
+        image.overlayWith(overlay, value.options);
+      } else if (key === "smartCrop") {
+        const options = value;
+        const imageBuffer = await image.toBuffer();
+        const metadata = await image.metadata();
+        // ----
+        const boundingBox = await this.getBoundingBox(
+          imageBuffer,
+          options.faceIndex
+        );
+        const cropArea = await this.getCropArea(boundingBox, options, metadata);
+        try {
+          image.extract(cropArea);
+        } catch (err) {
+          throw {
+            status: 400,
+            code: "SmartCrop::PaddingOutOfBounds",
+            message:
+              "The padding value you provided exceeds the boundaries of the original image. Please try choosing a smaller value or applying padding via Sharp for greater specificity."
+          };
+        }
+      } else if (key === "watermark") {
+        const raw = edits["watermark"].split(",");
+        const dimensions = {
+          width: parseInt(raw[0]),
+          height: parseInt(raw[1])
+        };
+        console.log(raw);
+        console.log(dimensions);
+        if (dimensions["height"] < 600) {
+          console.log("small");
+          image.overlayWith("logo.png", {
+            top: Math.round(dimensions["height"] * 0.2),
+            left: Math.round(dimensions["width"] * 0.1)
+          });
+        } else if (dimensions["height"] < 1200) {
+          console.log("medium");
+          image.overlayWith("logo_2x.png", {
+            top: Math.round(dimensions["height"] * 0.2),
+            left: Math.round(dimensions["width"] * 0.1)
+          });
+        } else {
+          console.log("large");
+          image.overlayWith("logo_3x.png", {
+            top: Math.round(dimensions["height"] * 0.2),
+            left: Math.round(dimensions["width"] * 0.1)
```

Comment:
Can we separate out the `overlayWith` function call from this if/else block? It's a lot of repeated code and looks like we can just have the choosing of the key in the if/else and call the function afterwards.